File: LinkedList/List Cycle(cycle begins).c
ListNode* Solution::detectCycle(ListNode* A) {

    if(A==NULL || A->next == NULL)
            return NULL;
        ListNode *s = A;
        ListNode *f = A;
        while(s!=NULL && f!=NULL && f->next!=NULL)  {
            s=s->next;
            f=f->next->next;
            if(s==f)
                break;
        }
        if(s!=f)
            return NULL;
            
        s = A; 
        while (s != f) { 
            s = s->next; 
            f = f->next; 
        }
    return s; 
}

File: LinkedList/List Cycle(cycle begins).java
/*
Given a linked list, return the node where the cycle begins. If there is no cycle, return null.
*/
public class Solution {
	public ListNode detectCycle(ListNode a) {
	    if(a==null || a.next == null)
	        return null;
	    
	    ListNode s = a;
	    ListNode f = a;
	    while(s!=null && f!=null && f.next!=null)
	    {
	        s=s.next;
	        f=f.next.next;
	        if(s==f)
	            break;
	    }
	    if(s!=f)
	        return null;
	        
	    s = a;
	    while(s!=f)
	    {
	        s=s.next;
            f=f.next;   
	    }
	    return s;
	}
}